server/src/main/java/org/opensearch/cluster/metadata/WeightedRoutingMetadata.java

```diff
@@ -99,7 +100,7 @@ public static NamedDiff<Metadata.Custom> readDiffFrom(StreamInput in) throws IOE
     public static WeightedRoutingMetadata fromXContent(XContentParser parser) throws IOException {
         String attrKey = null;
         Double attrValue;
-        String attributeName = null;
+        String attributeName = "";
```

Comment:
Why do we want to set it to empty string? Can we not just remove the custom metadata from cluster state if weighted routing is removed?

Reply:
Can we not handle this with a defensive null check while reads and writes instead and marking attributeName as Optional to make this more explicit

Reply:
> Can we not handle this with a defensive null check while reads and writes instead and marking attributeName as Optional to make this more explicit

Yeah makes sense. Added defensive checks when reading/writing values. Even if we make `attributeName` as optional, it has to be initialized with a value which in this case is empty since the metadata is set [empty](https://github.com/opensearch-project/OpenSearch/blob/13e05b5d2a9edf2335f582bb340391a51dea07f2/server/src/main/java/org/opensearch/cluster/metadata/WeightedRoutingMetadata.java#L157) if weights are deleted.  Don't see any additional benefit of going with Optional. Let me know if I am missing anything

Reply:
Since we are adding null check any reason to keep it as empty string?

Reply:
I have removed the null check. Its kind of just hiding the problem.

Reply:
We are initializing `attributeName` with empty string in the [constructor](https://github.com/opensearch-project/OpenSearch/blob/456038bf201bc8344952081e51b0812eb76aa0e9/server/src/main/java/org/opensearch/cluster/routing/WeightedRouting.java#L32) too.  This change will ensure consistent state